src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityResource.groovy

```diff
@@ -0,0 +1,87 @@
+package edu.oregonstate.mist.directoryapi
+
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.AuthenticatedUser
+import io.dropwizard.auth.Auth
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.ResponseBuilder
+import javax.ws.rs.core.MediaType
+
+/**
+ * DirectoryEntity Resource class.
+ */
+@Path('/directory')
+class DirectoryEntityResource extends Resource {
+    private final DirectoryEntityDAO directoryEntityDAO
+
+    /**
+     * Constructs the object after receiving and storing directoryEntityDAO instance.
+     */
+    public DirectoryEntityResource(DirectoryEntityDAO directoryEntityDAO) {
+        this.directoryEntityDAO = directoryEntityDAO
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getByParameters(
+            @Auth AuthenticatedUser authenticatedUser,
```

Comment:
In other words, I would call this api using: /directory?q=Smith and it would search ldap for all the records where one of the field names matches Smith. I could just as well use: /directory?q=cedenoj and it would find my record since that's my username.